Modules/CalcOffence-3_0.lua

```diff
@@ -2002,6 +2008,42 @@ function calcs.offence(env, actor, activeSkill)
 				t_insert(breakdown.EnemyStunDuration, s_format("= %.2fs", output.EnemyStunDuration))
 			end
 		end
+
+        -- Calculate impale chance and modifiers
+		if canDeal.Physical and output.ImpaleChance > 0 then
+            skillFlags.impale = true
+            local impaleChance = output.ImpaleChance/100
```

Comment:
```suggestion
            local impaleChance = m_min(output.ImpaleChance/100, 1)
```
Dunno if it's necessary with `combineStat` below, but you may wanna cap at 100% chance.